### ISSUE TITLE
Store files somewhere beside /tmp?

### ISSUE THREAD

**Issue by timholy** [JuliaSmoothOptimizers/CUTEst.jl#103]
Every time I reboot my laptop (e.g., after security updates that affect the kernel) I need to do `Pkg.rebuild("CUTEst")`. Any thoughts on storing the files elsewhere?

**Comment by dpo** [comment]
What's in /tmp that requires a rebuild?

**Comment by timholy** [comment]
`/tmp` gets wiped when you reboot, and
```sh
$ cat cutestenv.jl 
PATH = get(ENV, "PATH", "")
MANPATH = get(ENV, "MANPATH", "")
LD_LIBRARY_PATH = get(ENV, "LD_LIBRARY_PATH", "")
ENV["ARCHDEFS"] = "/tmp/tmplJ4BFj/archdefs"
ENV["CUTEST"] = "/tmp/tmplJ4BFj/cutest"
ENV["SIFDECODE"] = "/tmp/tmplJ4BFj/sifdecode"
ENV["MASTSIF"] = "/tmp/tmplJ4BFj/mastsif"
ENV["MYARCH"] = "pc64.lnx.gfo"
ENV["PATH"] = "/tmp/tmplJ4BFj/cutest/bin:/tmp/tmplJ4BFj/sifdecode/bin:$PATH"
ENV["MANPATH"] = "/tmp/tmplJ4BFj/cutest/man:/tmp/tmplJ4BFj/sifdecode/man:$MANPATH"
ENV["LD_LIBRARY_PATH"] = "/tmp/tmplJ4BFj/lib:$LD_LIBRARY_PATH"
```

**Comment by timholy** [comment]
Perhaps this is an unintentional consequence of the build script writing its log files to a directory created by `mktempdir`? I naively assumed this was intentional.

**Comment by abelsiqueira** [comment]
Really not intentional. Looks like a result of [this line](https://github.com/JuliaSmoothOptimizers/CUTEst.jl/blob/develop/deps/build.jl#L11).

**Comment by timholy** [comment]
I'll fix. I'm also curious about [this line](https://github.com/JuliaSmoothOptimizers/CUTEst.jl/blob/c69f7eeb810d09327b25d08b2f53506b0d328b4a/deps/build.jl#L35). Why delete the file if everything is already installed?

**Comment by dpo** [comment]
> Really not intentional. Looks like a result of this line.

Ah I guess I naively thought `cd` would "come back" when the function exits.

> I'm also curious about this line. Why delete the file if everything is already installed?

The idea is that if `validate_cutest()` returns successfully, a pre-existing installation of CUTEst has been found (outside of Julia, e.g., with Homebrew). In that case `cutestenv.jl` should be removed because the environment variables in it may not point to the location of the working installation.

**Comment by timholy** [comment]
> I naively thought cd would "come back" when the function exits.

It does if you use the do-block form.

> In that case cutestenv.jl should be removed because the environment variables in it may not point to the location of the working installation.

Ah, thanks for explaining.

See #104.

**Comment by abelsiqueira** [comment]
Closed by #104
